examples/cosmicink.lua

```diff
@@ -0,0 +1,645 @@
+			bashls = "sh",
+			jsonls = "json",
+			html = "html",
+			cssls = "css",
+			tailwindcss = "tw",
+			dockerls = "docker",
+			sqlls = "sql",
+			yamlls = "yml",
+		}
+		for _, client in ipairs(clients) do
+			local filetypes = client.config.filetypes
+			if filetypes and vim.fn.index(filetypes, buf_ft) ~= -1 then
+				return lsp_short_names[client.name] or client.name:sub(1, 2)
+			end
+		end
+		return msg
+	end,
+	icon = ' ',
+	color = {
+		fg = colors.YELLOW,
+		gui = 'bold',
+	},
+}
+
+ins_right {
+	function()
+		return ''
+	end,
+	color = function()
+		return { fg = get_middle_color() }
+	end,
+	cond = hide_in_width,
+}
+
+ins_right(
+	create_separator('right')
+)
+
+ins_right(
+	create_mode_based_component('location', nil, colors.BG)
+)
+
+ins_right(
+	create_separator('left')
+)
+
+ins_right {
+	'branch',
+	icon = ' ',
+	--[[ Truncates and formats Git branch names for display in lualine:
+    First segment: Uppercase, truncated to 1 character.
+    Middle segments: Lowercase, truncated to 1 character.
+    Last segment: Unchanged.
+    Separator: › between truncated segments and the last segment.
+
+    Example Input/Output:
+		Branch										Name	Output
+		backend/setup/tailwind		Bs›tailwind
+		feature/add-ui						Fa›add-ui
+		main											main
+	]]
+	fmt = function(branch)
+		if branch == '' or branch == nil then
+			return 'No Repo'
+		end
+
+		-- Function to truncate a segment to a specified length
+		local function truncate_segment(segment, max_length)
+			if #segment > max_length then
+				return segment:sub(1, max_length)
+			end
+			return segment
+		end
+
+		-- Split the branch name by '/'
+		local segments = {}
+		for segment in branch:gmatch('[^/]+') do
+			table.insert(segments, segment)
+		end
+
+		-- Truncate all segments except the last one
+		for i = 1, #segments - 1 do
+			segments[i] = truncate_segment(segments[i], 1) -- Truncate to 1 character
+		end
+
+		-- If there's only one segment (no '/'), return it as-is
+		if #segments == 1 then
+			return segments[1]
+		end
+
+		-- Capitalize the first segment and lowercase the rest (except the last one)
+		segments[1] = segments[1]:upper() -- First segment uppercase
+		for i = 2, #segments - 1 do
+			segments[i] = segments[i]:lower() -- Other segments lowercase
+		end
+
+		-- Combine the first segments with no separator and add '›' before the last segment
+		local truncated_branch = table.concat(segments, '', 1, #segments - 1) .. '›' .. segments[#segments]
+
+		-- Ensure the final result doesn't exceed a maximum length
+		local max_total_length = 15
+		if #truncated_branch > max_total_length then
+			truncated_branch = truncated_branch:sub(1, max_total_length) .. '…'
+		end
+
+		return truncated_branch
+	end,
+	color = function()
+		local mode_color = get_mode_color()
+		return {
+			fg = mode_color,
+			gui = 'bold',
+		}
+	end,
+}
+
+ins_right(
+	create_separator('right')
+)
+
+ins_right(
+	create_mode_based_component('progress', nil, colors.BG)
+)
+
+return {
```

Comment:
You should directly call require'lualine'.setup(config) at end of the file instead of needing lazy.nvim to use this.